src/leaderboard/messages/data/fetch_by_rank.rs

```diff
@@ -0,0 +1,88 @@
+use super::{fetch::FetchResponse, Order, RankedElement};
+use crate::leaderboard::MomentoRequest;
+use crate::utils::prep_leaderboard_request_with_timeout;
+use crate::{Leaderboard, MomentoResult};
+
+use std::ops::Range;
+
+/// Represents a range of ranks used to request elements by rank.
+pub struct RankRange {
+    start_inclusive: u32,
+    end_exclusive: u32,
+}
+
+impl From<Range<u32>> for RankRange {
+    fn from(val: std::ops::Range<u32>) -> Self {
+        RankRange {
+            start_inclusive: val.start,
+            end_exclusive: val.end,
+        }
+    }
+}
+
+impl From<RankRange> for momento_protos::leaderboard::RankRange {
+    fn from(val: RankRange) -> Self {
+        momento_protos::leaderboard::RankRange {
+            start_inclusive: val.start_inclusive,
+            end_exclusive: val.end_exclusive,
+        }
+    }
+}
+
+/// A request to get ranked elements by rank.
+pub struct FetchByRankRequest {
+    rank_range: RankRange,
+    order: Order,
+}
+
+impl FetchByRankRequest {
+    /// Constructs a new `FetchByRankRequest`.
+    pub fn new<T: Into<RankRange>>(rank_range: T) -> Self {
+        Self {
+            rank_range: rank_range.into(),
+            order: Order::Ascending,
+        }
+    }
+
+    /// Sets the order of the elements to be fetched.
+    pub fn order(mut self, order: Order) -> Self {
+        self.order = order;
+        self
+    }
```

Comment:
somewhere, or in both places, this should specify what the ordering default is

Reply:
Clarified in [c5800e6](https://github.com/momentohq/client-sdk-rust/pull/438/commits/c5800e6c8a0e9be406c6a1b4abc416c1f6fc1ad1)